runtimevar/httpvar/httpvar.go

```diff
@@ -0,0 +1,236 @@
+	return fmt.Sprintf("httpvar: recived status code %d while calling \"%s\"", e.StatusCode, e.URL)
+}
+
+func newRequestError(statusCode int, url string) *RequestError {
+	return &RequestError{statusCode, url}
+}
+
+// New constructs a *runtimevar.Variable that uses http.Client to watch the variable
+// name on an http endpoint.
+func NewVariable(client *http.Client, endpoint string, decoder *runtimevar.Decoder, opts *Options) (*runtimevar.Variable, error)  {
+	if client == nil {
+		client = http.DefaultClient
+	}
+
+	endpointUrl, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("httpvar: fail to parse url: %s", err.Error())
+	}
+
+	return runtimevar.New(newWatcher(client, endpointUrl, decoder, opts)), nil
+}
+
+type state struct {
+	val        interface{}
+	raw        *http.Response
+	rawBytes   []byte
+	updateTime time.Time
+	err        error
+}
+
+// Value implements driver.State.Value.
+func (s *state) Value() (interface{}, error) {
+	return s.val, s.err
+}
+
+// UpdateTime implements driver.State.UpdateTime.
+func (s *state) UpdateTime() time.Time {
+	return s.updateTime
+}
+
+// As implements driver.State.As.
+func (s *state) As(i interface{}) bool {
+	if s.raw == nil {
+		return false
+	}
+	p, ok := i.(**http.Response)
+	if !ok {
+		return false
+	}
+	*p = s.raw
+	return true
+}
+
+// errorState returns a new State with err, unless prevS also represents
+// the same error, in which case it returns nil.
+func errorState(err error, prevS driver.State) driver.State {
+	s := &state{err: err}
+	if prevS == nil {
+		return s
+	}
+	prev := prevS.(*state)
+	if prev.err == nil {
+		// New error.
+		return s
+	}
+	if equivalentError(err, prev.err) {
+		// Same error, return nil to indicate no change.
+		return nil
+	}
+	return s
+}
+
+// equivalentError returns true if err1 and err2 represent an equivalent error;
+// i.e., we don't want to return it to the user as a different error.
+func equivalentError(err1, err2 error) bool {
+	if err1 == err2 || err1.Error() == err2.Error() {
+		return true
+	}
+	var code1, code2 int
+	if e, ok := err1.(*RequestError); ok {
+		code1 = e.StatusCode
+	}
+	if e, ok := err2.(*RequestError); ok {
+		code2 = e.StatusCode
+	}
+	return code1 != 0 && code1 == code2
+}
+
+// watcher implements driver.Watcher for configurations provided by the Runtime Configurator
+// service.
+type watcher struct {
+	client   *http.Client
+	endpoint *url.URL
+	decoder  *runtimevar.Decoder
+	wait     time.Duration
+}
+
+// WatchVariable implements driver.WatchVariable.
+func (w *watcher) WatchVariable(ctx context.Context, prev driver.State) (driver.State, time.Duration) {
+	resp, err := w.client.Get(w.endpoint.String())
+	if err != nil {
+		return errorState(err, prev), w.wait
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		err := newRequestError(resp.StatusCode, w.endpoint.String())
+		return errorState(err, prev), w.wait
+	}
+
+	respBodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
I'm not sure how to trigger an error from `ioutil.ReadAll`, looking for your ideas on how to test this